### ISSUE TITLE
useAxiosRepo with typescript - Argument of type 'typeof MyModel' is not assignable to parameter of type 'Model'

### ISSUE THREAD

**Issue by nowrap** [CodeDredd/pinia-orm#1795]
Hello,

i am using the axios plugin in a typescript project to retrive data from an api.
Running the code in dev mode works. But it won't compile:

```
Argument of type 'typeof MyModel' is not assignable to parameter of type 'Model'.
  Type 'typeof MyModel' is missing the following properties from type 'Model': _meta, $self, $entity, $config, and 38 more.

await useAxiosRepo(MyModel).api().get('mymodels', {
```

Maybe someone has a suggestion for me

Regards
nowrap

**Comment by nowrap** [comment]
The useAxiosRepo definition is:
```
declare function useAxiosRepo<M extends Model>(model: M): AxiosRepository<Model>;
```

And the useRepo definition is:
```
declare function useRepo<R extends Repository>(repository: R | Constructor<R>, pinia?: Pinia): R;
declare function useRepo<M extends Model>(model: Constructor<M>, pinia?: Pinia): Repository<M>;
```

**Comment by i2pacg** [comment]
Any solution or workaround to this , it seems to work while still showing the error.

**Comment by CodeDredd** [comment]
It should be fixed in the edge channel and later in the 1.9.0 release. Feel free to test it out.
Feedback always welcome.

**Comment by CodeDredd** [comment]
Fixed with #816
